### PR TITLE
Setup readthedocs

### PR DESCRIPTION
**Pull request by Jean-Francois-Lafleche**
### Description
Add support for readthedocs documentation builds

### Tests
OS: Ubuntu 18.04
Driver: 440.36
CUDA: 10.2
Python 3.6

Readthedocs - Confirmed documentations are successfully built from fork branch.

Commands:
- pytest tests/

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def read(*names, **kwargs):
     )
 
     import torch
-    from torch.utils.cpp_extension import BuildExtension, CUDAExtension
+    from torch.utils.cpp_extension import BuildExtension, CUDAExtension, CUDA_HOME
```
Comment (krrish94) [inline setup.py]: Curious as to what this import does.

File: setup.py [hunk 2]
```diff
@@ -136,43 +136,47 @@ def read(*names, **kwargs):
...
-            'kaolin/graphics/nmr/cuda/rasterize_cuda_kernel.cu',
-        ]),
         triangle_hash_module,
         mise_module,
         mcubes_module,
         nnsearch_module,
     ]
+    
+    # If building with readthedocs, don't compile CUDA extensions
+    if os.getenv('READTHEDOCS') != 'True':
```
Comment (krrish94) [inline setup.py:146]: Is this the recommended way to check if we're building on readthedocs?

**Review by krrish94 — COMMENTED** [review]
CUDA_HOME appears to be unused.

And, I'm not 100% sure if the check for an envvar `READTHEDOCS` is the recommended way.

File: setup.py (hunk 1 above)
Comment (Jean-Francois-Lafleche) [reply setup.py]: Ah, you're right. I can take that out now. I used it in an earlier version when checking for whether the local machine has CUDA installed and setup properly. It tries to find the CUDA install path: https://pytorch.org/docs/stable/_modules/torch/utils/cpp_extension.html#CppExtension (see _find_cuda_home()

File: setup.py (hunk 2 above)
Comment (Jean-Francois-Lafleche) [reply setup.py:146]: Not sure if there is a recommended way but happy to change it if you have a alternative suggestion. This works well and can be easily extended with a CUDA check once we support CPU-only installation.

Comment (Jean-Francois-Lafleche) [reply setup.py:146]: Ah, it appears this is the recommended way afterall: https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-behavior-for-read-the-docs

File: .readthedocs.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+version: 2
+
+python:
+   version: 3.7
```
Comment (Caenorst) [inline .readthedocs.yml]: Can we use python 3.6 as we have problems with 3.7 on kaolin ? It avoid us to have multiple python versions.

File: docs/conf.py [hunk 4]
```diff
@@ -12,6 +12,16 @@
 
 import os
 import sys
+
+
+# Get version number
+version_path = os.path.join('../', 'kaolin', 'version.py')
```
Comment (Caenorst) [inline docs/conf.py]: why not just `version_path = '../kaolin/version.py' ?

File: docs/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+-f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+numpy
+torch_nightly
```
Comment (Caenorst) [inline docs/requirements.txt]: Do we really want to have dependency on torch_nightly ? Shouldn't we agree on a release instead ?

**Review by Caenorst — COMMENTED** [review]
Just few questions to answer, otherwise LGTM

File: docs/requirements.txt (hunk 5 above)
Comment (Jean-Francois-Lafleche) [reply docs/requirements.txt]: Fixed. Now using pinned pytorch version 1.3.1 with corresponding torchvision version.

File: docs/conf.py (hunk 4 above)
Comment (Jean-Francois-Lafleche) [reply docs/conf.py]: Fixed. You are absolutely correct.

File: .readthedocs.yml (hunk 3 above)
Comment (Jean-Francois-Lafleche) [reply .readthedocs.yml]: Fixed. Python 3.7 issues should be resolved with PR #66, but fixing to python 3.6 in any case as it is the development version and is more regularly tested.

**Pull request merged**